cola/linalg/inv.py

```diff
@@ -168,3 +168,16 @@ def inv(A: Diagonal, **kwargs):
 @dispatch
 def inv(A: Triangular, **kwargs):
     return TriangularInv(A)
+
+
+@dispatch
+def inv(A: Sum[Product[Dense,Dense], Diagonal], **kwargs):
```

Comment:
We should be able to generalize this to
`Sum[Product[LinearOperator,LinearOperator]` right?

Also I believe we should be able to generalize `Diagonal` to `efficiently_invertible = Union[Diagonal, Identity, ScalarMul]`, we could extend that further to `Triangular, Permutation`.